src/runtime/CollectionWrappers/IterableWrapper.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class IterableWrapper<T> : IEnumerable<T>
+    {
+        protected readonly PyObject pyObject;
+
+        public IterableWrapper(PyObject pyObj)
+        {
+            if (pyObj == null)
+                throw new PythonException();
+            pyObject = pyObj;
+        }
+
+        private void propagateIterationException()
+        {
+            var err = Runtime.PyErr_Occurred();
+            if (err == null) return;
+
+            //remove StopIteration exceptions
+            if (0 != Runtime.PyErr_ExceptionMatches(Exceptions.StopIteration))
+            {
+                Runtime.PyErr_Clear();
+                return;
+            }
```

Comment:
According to the official documentation, `PyIter_Next` should not be setting `StopIteration` exception:
https://docs.python.org/3.5/c-api/iter.html#c.PyIter_Next
> If there are no remaining values, returns NULL with no exception set